api/src/main/java/io/minio/MinioClient.java

```diff
@@ -1938,7 +1940,8 @@ public void makeBucket(String bucketName, String region)
       configString = config.toString();
     }
 
-    executePut(bucketName, null, headerMap, null, US_EAST_1, configString, 0);
+    execute(Method.PUT, US_EAST_1, bucketName, null, headerMap, null,
+            "application/octet-stream", configString, 0);
```

Comment:
What i am saying its confusing for newer patches. Nitish tried to use executePut for copyObject it didn't work and then had to use lower level execute() function - what i am saying is abstraction has a limit and it is not addressing all scenarios so it breaking. 

It is better for each caller to know if and when they need to close. Closing internally and not knowing  that it would be closed is hard to follow. For example executeGet and executePost don't close the body underneath - all abstracted wrappers for http methods should have same behavior otherwise we won't remember.